NewRelicBrowser.php

```diff
@@ -1,392 +1,456 @@
-							'timeout' => 100,
-							'method' => 'POST',
-							'headers' => array(
-								'x-api-key' => RTP_NEW_RELIC_API_KEY,
-								'Content-Type' => 'application/json',
-							),
-							'body' => $dataString,
-								) );
+							);
+
+							/* for this api data is to be pass in json */
+							$datastring = wp_json_encode( $data );
+
+							$create_browser_response = wp_remote_post(
+								RTP_NEW_RELIC_API_URL,
+								array(
+									'timeout' => 100,
+									'method'  => 'POST',
+									'headers' => array(
+										'x-api-key'    => RTP_NEW_RELIC_API_KEY,
+										'Content-Type' => 'application/json',
+									),
+									'body'    => $datastring,
+								)
+							);
+
 							$create_browser_response_code = wp_remote_retrieve_response_code( $create_browser_response );
+
 							$json_data = json_decode( $create_browser_response['body'] );
-							if ( 201 != $create_browser_response_code ) {
+							if ( 201 !== (int) $create_browser_response_code ) {
 								add_settings_error( 'relic_options', 'relic_options_error', $json_data->error, 'error' );
-							} else if ( isset( $json_data->api_key ) ) {
+							} elseif ( isset( $json_data->api_key ) ) {
 								/* store the received data */
 								$main_array = array(
 									'relic_account_name' => $json_data->name,
-									'relic_api_key' => $json_data->api_key,
-									'relic_id' => $json_data->id,
-									'relic_password' => $relic_password,
+									'relic_api_key'      => $json_data->api_key,
+									'relic_id'           => $json_data->id,
+									'relic_password'     => $relic_password,
 								);
 								add_option( $option_name, $main_array );
-									/* end of API 1
-								  Now create the browser app */
+									/**
+									 * End of API 1
+									 * Now create the browser app
+									 */
 								if ( isset( $_POST['relic-account-name'] ) ) {
-									$relic_account_name = sanitize_text_field( $_POST['relic-account-name'] );
+									$relic_account_name = sanitize_text_field( wp_unslash( $_POST['relic-account-name'] ) );
 								}
 								$browser_created = $this->rtp_create_browser_app( $relic_account_name, $json_data->api_key );
 								if ( $browser_created ) {
 									/* mail the account details to user */
 									if ( isset( $_POST['relic-account-email'] ) ) {
-										$relic_user_mail = sanitize_email( $_POST['relic-account-email'] );
+										$relic_user_mail = sanitize_email( wp_unslash( $_POST['relic-account-email'] ) );
+
 										$browser_app_details = get_option( 'rtp_relic_browser_details' );
+
 										$relic_subject = 'Welcome to New Relic Browser monitoring of ' . $relic_account_name;
-										$relic_headers = "MIME-Version: 1.0\r\n";
+
+										$relic_headers  = "MIME-Version: 1.0\r\n";
 										$relic_headers .= "Content-Type: text/html; charset=ISO-8859-1\r\n";
+
 										$relic_email_message = '<div style="font-size:15px;margin-top:20px;border:1px solid #666;padding:20px 50px;">
```

Comment:
Move HTML to separate template file and then use it here.

Reply:
Also, escape `$relic_account_name`, `$relic_user_mail` and etc variables in mail message.

Reply:
i solved this issue. please review changes.